server/resource_server/service/service_test.go

```diff
@@ -1059,3 +1064,36 @@ func TestRegisterUserv2_Success(t *testing.T) {
 
 	assert.Nil(t, err)
 }
+
+func TestUpdateUserPasswordV2_Success(t *testing.T) {
+	mockRepo := &mockRepository{
+		updateUserPasswordV2: func(username, storedKey, serverKey string) error {
+			assert.Equal(t, "test_user", username)
+			assert.Equal(t, "test_stored_key", storedKey)
+			assert.Equal(t, "test_server_key", serverKey)
```

Comment:
Here you should also move hard-coded strings into separate constant variables which you can define on the top level (to reuse them in the next test as well).